packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -86,4 +90,28 @@ class UpgradeCommand extends FlutterCommand {
       || _gitChangedRegex.hasMatch(line)
       || line == 'Fast-forward';
   }
+
+  /// Find and return the project root directory relative to the specified
+  /// directory or the current working directory if none specified.
+  /// Return `null` if the project root could not be found
+  /// or if the project root is inside the flutter repository.
```

Comment:
I think it should be ok to `upgrade` in a project inside flutter root? For instance, `//examples/flutter_gallery` should be legal.